lib/core/CMakeLists.txt

```diff
@@ -1,34 +1,41 @@
 # This file is part of covfie, a part of the ACTS project
 #
-# Copyright (c) 2022-2023 CERN
+# Copyright (c) 2022-2024 CERN
 #
 # This Source Code Form is subject to the terms of the Mozilla Public License,
 # v. 2.0. If a copy of the MPL was not distributed with this file, You can
 # obtain one at http://mozilla.org/MPL/2.0/.
 
-add_library(core INTERFACE)
+add_library(covfie_core INTERFACE)
 
 target_include_directories(
-    core
+    covfie_core
     INTERFACE
         $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}>
         $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
 )
 
-target_compile_features(core INTERFACE cxx_std_20)
+target_compile_features(covfie_core INTERFACE cxx_std_20)
 
 if(COVFIE_QUIET)
-    target_compile_definitions(core INTERFACE COVFIE_QUIET)
+    target_compile_definitions(covfie_core INTERFACE COVFIE_QUIET)
 endif()
 
 # Logic to ensure that the core module can be installed properly.
-install(TARGETS core EXPORT ${PROJECT_NAME}Targets)
+set_target_properties(
+    covfie_core
+    PROPERTIES
+        EXPORT_NAME
+            core
+)
+
+install(TARGETS covfie_core EXPORT ${PROJECT_NAME}Targets)
```

Comment:
Do we really need to make this so complicated? Can't we just set `install(... NAMESPACE something_)`?

Reply:
I didn't touch the code that specifies the `NAMESPACE`. The `EXPORT_NAME` needs to be specified, because without it, the exported library would now be called `covfie::covfie_core`.

This is exactly how the "trick" of using `vecmem::core`, `detray::core`, etc. is achieved, whether or not we're using the project directly, or we're finding an installed version of it. (With the "actual target names" being `vecmem_core`, `detray_core`, etc.)

Reply:
Right, but this is what I am saying, just like Google Benchmark does it, the target should be named e.g. `core` and then the namespace should be added to it, so that the full name becomes `covfie::core`.